### ISSUE TITLE
Use masked mean for pooled scene encoding

### ISSUE THREAD

**Issue by Max-Bin** [tier4/Diffusion-Planner#187]
## Problem

`diffusion_planner/diffusion_planner/model/module/decoder.py` currently computes pooled scene encoding with a plain mean over the full token dimension:

```python
encoding_pooled = torch.mean(encoding, dim=1)  # [B, D]
```

This averages over all fixed encoder token slots, including padded / invalid slots. As a result, the pooled representation depends on the number of allocated token slots, not only on the actual valid scene content.

This is especially risky after increasing the maximum number of neighbor slots, because the same scene can have many more empty neighbor tokens. Even if those invalid tokens are zeroed, including them in the denominator reduces the magnitude of `encoding_pooled`.

## Impact

`encoding_pooled` is used by the turn-indicator prediction path together with the ego trajectory. If the number of padded tokens changes, the input scale of the turn-indicator head also changes.

For example, increasing the neighbor capacity from 32 to 320 can add hundreds of empty slots. For sparse scenes, this can substantially dilute the pooled encoding and create a distribution shift for the turn-indicator predictor.

This does not directly corrupt the trajectory training target, but it can weaken scene-level conditioning and turn-indicator prediction quality. The issue is mainly a scale / distribution-shift bug introduced by fixed-size token expansion.

## Expected behavior

The pooled encoding should be computed as a masked mean over valid encoder tokens only:

```python
encoding_pooled = (encoding * valid_mask[..., None]).sum(dim=1) / valid_mask.sum(dim=1).clamp_min(1)[..., None]
```

The valid-token mask should follow the same semantic convention as the encoder mask so that padded tokens are excluded from the denominator.

## Notes

A fix should preserve behavior when all tokens are valid, and should add a regression check that changing only the number of padded neighbor slots does not change the pooled encoding scale.
